plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeSource.java

```diff
@@ -36,16 +36,20 @@
 public class FileSystemExchangeSource
         implements ExchangeSource
 {
+    private final FileSystemExchangeStats stats;
     private final List<ExchangeStorageReader> readers;
+    private volatile CompletableFuture<Void> blocked;
```

Comment:
Is this an optimization? Why do we need a class member `blocked`?

Reply:
This is to make sure only a single `blocked` future exists and tracked.

Reply:
I don't quite understand, why not simply the following:
```
return stats.getExchangeSourceBlocked().record(toCompletableFuture(
                nonCancellationPropagating(
                        whenAnyComplete(readers.stream()
                                .map(ExchangeStorageReader::isBlocked)
                                .collect(toImmutableList())))));
```